web/files/referencedby.go

```diff
@@ -10,6 +11,36 @@ import (
 	"github.com/labstack/echo/v4"
 )
 
+// ReferencesDocument is JSON-API document, identified by the mediatype
+// application/vnd.api+json
+// See http://jsonapi.org/format/#document-structure
+// It differs from jsonapi.Document because it represents a list of relationships
+// but differs from jsonapi.Relationship because it olds the revision of it
+// parent io.cozy.files document.
+type ReferencesDocument struct {
+	Data *json.RawMessage `json:"data"`
+	Meta jsonapi.Meta     `json:"meta"`
+}
+
+// WriteReferences can be called to write an answer with a JSON document
+// containing the list of added reference_by objects as data into an io.Writer.
+func WriteReferences(c echo.Context, newRev string, refs []couchdb.DocReference) error {
```

Comment:
I have read again the `pkg/jsonapi` package. There is a `DataRelations` function that is doing something really close to this function. I think we should modify it a bit to accept the revision in the meta.

Reply:
I saw `DataRelations` but it didn't have the `rev` field and I didn't want to modify this package.
Adding the `rev` field to `RelationshipMeta` would do the trick.